### PR TITLE
Lower final switch default error to object.d template

### PR DESCRIPTION
**Pull request by somzzz**
Druntime changes: https://github.com/dlang/druntime/pull/1971

I still had to use ```SwitchErrorStatement``` because of the checks and actions the compiler performs for ```ErrorStatemenets``` vs regular ```Statements```.

### CODE REVIEW

File: src/ddmd/backend/rtlsym.d [hunk 1]
```diff
@@ -53,9 +53,9 @@ enum
     RTLSYM_MONITOREXIT,
     RTLSYM_CRITICALENTER,
     RTLSYM_CRITICALEXIT,
-    RTLSYM_SWITCH_STRING,
-    RTLSYM_SWITCH_USTRING,
-    RTLSYM_SWITCH_DSTRING,
+    RTLSYM_SWITCH_STRING,       // unused
+    RTLSYM_SWITCH_USTRING,      // unused
+    RTLSYM_SWITCH_DSTRING,      // unused
```
Comment (wilzbach) [inline src/ddmd/backend/rtlsym.d]: https://issues.dlang.org/show_bug.cgi?id=9395 would be very useful here ;-)

File: src/ddmd/s2ir.d [hunk 2]
```diff
@@ -691,19 +691,16 @@ extern (C++) class S2irVisitor : Visitor
...
+            //printf("SwitchErrorStatement.toIR(), exp = %s\n", s.exp ? s.exp.toChars() : "");
+            incUsage(irs, s.loc);
+            block_appendexp(blx.curblock, toElemDtor(s.exp, irs));
         }
         else
         {
-            auto efilename = el_ptr(toSymbol(cast(Dsymbol)blx._module));
-            auto elinnum = el_long(TYint, s.loc.linnum);
-            e = el_bin(OPcall, TYvoid, el_var(getRtlsym(RTLSYM_DSWITCHERR)), el_param(elinnum, efilename));
+            assert(0);
```
Comment (wilzbach) [inline src/ddmd/s2ir.d]: It's a good style to provide an informative error message or do sth. like 

```d
assert(s.exp !is null, "SwitchErrorStatement needs to have a valid Expression.") 
```

and skip the `if` entirely.

**Review by wilzbach — COMMENTED** [review]
Nits

File: src/ddmd/s2ir.d (hunk 2 above)
Comment (JinShil) [reply src/ddmd/s2ir.d]: > It's a good style to provide an informative error message or do sth. like...

@WalterBright disagrees: https://github.com/dlang/dmd/pull/7079#discussion_r132904181

> Some people like assert messages, I don't. They bloat up the compiler. Leaving the message as a comment in the source code is plenty good enough.

I really don't mind either way, just an FYI.

**Review by andralex — APPROVED** [review]

**Pull request merged**